src/emulator/functionsEmulator.ts

```diff
@@ -742,39 +763,50 @@ export class FunctionsEmulator implements EmulatorInstance {
     return Object.values(this.triggers).map((record) => record.def);
   }
 
-  getTriggerDefinitionByKey(triggerKey: string): EmulatedTriggerDefinition {
+  getTriggerRecordByKey(triggerKey: string): EmulatedTriggerRecord {
```

Comment:
Aside - Keeping track of using this fn to get `backend` associated w/ a trigger and sometimes relying on passed `backend` gets difficult. I __think__ this is b/c we use `startFunctionsRuntime` to both parse triggers (which means that we don't have a trigger record yet) and to invoke a functio n(which means that we do have the trigger record).

Maybe this is something I'll try to clarify when it comes time to implement container contract.

Reply:
+100. I think cleanly separating trigger discovery and request handling/code execution is one of the biggest improvements we can make to the readability of this code.